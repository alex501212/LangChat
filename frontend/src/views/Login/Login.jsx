import React, { useState } from "react";
import "./Login.scss";
import { Flex, Input, Heading, Button, useToast } from "@chakra-ui/react";

const Login = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const toast = useToast();

  const signIn = () => {
    if(username !== "") {
    fetch("http://localhost:5000/login", {
      method: "POST",
      headers: {
        "Content-type": "application/json",
      },
      body: JSON.stringify({
        username: username,
        password: password,
      }),
    })
      .then((res) => res.json())
      .then((data) => {
        if (data.status === "ok") {
          sessionStorage.setItem("token", data.token);
          window.location.replace("/dashboard");
        } else {
          toast({
            description: data.message,
            status: "error",
            duration: 9000,
            isClosable: true,
            containerStyle: {
              width: "500px",
              maxWidth: "100%"
            },
          });
        }
      });
    } else {
      toast({
        description: "Invalid username or password",
        status: "error",
        duration: 1000,
        isClosable: true,
      });
    }
  };

  return (
    <Flex
      height="90vh"
      justifyContent="center"
      alignItems="center"
      background="gray.500"
    >
      <Flex
        direction="column"
        p={10}
        borderWidth="5px"
        borderRadius="lg"
        background="white"
      >
        <Heading mb={10}>Log in to LangChat</Heading>
        <Input
          value={username}
          onChange={(e) => setUsername(e.target.value)}
          onKeyDown={(e) => {
            if (e.key === "Enter") {
              signIn();
            }
          }}
          mb={5}
          placeholder="Username"
        />
        <Input
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          onKeyDown={(e) => {
            if (e.key === "Enter") {
              signIn();
            }
          }}
          mb={5}
          type={"password"}
          placeholder="Password"
        />

        <Button type="submit" onClick={() => signIn()} colorScheme="blue">
          Login
        </Button>
      </Flex>
    </Flex>
  );
};

export default Login;
