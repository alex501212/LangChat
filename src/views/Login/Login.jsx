import React, { useState } from "react";
import "./Login.scss";
import {
  Flex,
  Input,
  Heading,
  Button,
  useToast,
  InputGroup,
  InputRightElement,
} from "@chakra-ui/react";
import bg from "../../bg.png";

const Login = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [showPass, setShowPass] = useState(false);

  const toast = useToast();

  const handleHideToggle = () => setShowPass(!showPass);

  const signIn = () => {
    if (username !== "") {
      fetch("https://langchat-api.onrender.com/login", {
        method: "POST",
        headers: {
          "Content-type": "application/json",
        },
        body: JSON.stringify({
          username: username,
          password: password,
        }),
      })
        .then((res) => res.json())
        .then((data) => {
          if (data.status === "ok") {
            sessionStorage.setItem("token", data.token);
            window.location.replace("/dashboard");
          } else {
            toast({
              description: data.error,
              status: "error",
              duration: 9000,
              isClosable: true,
              containerStyle: {
                width: "500px",
                maxWidth: "100%",
              },
            });
          }
        });
    } else {
      toast({
        description: "Invalid username or password",
        status: "error",
        duration: 1000,
        isClosable: true,
      });
    }
  };

  return (
    <Flex
      boxShadow="inner"
      height="90vh"
      justifyContent="center"
      alignItems="center"
      style={{
        background: `url(${bg})`,
        backgroundSize: "1920px 940px",
        backgroundPosition: "0px -94px",
        backgroundColor: "#718096"
      }}
    >
      <Flex
        direction="column"
        p={10}
        borderWidth="5px"
        borderRadius="lg"
        background="white"
        boxShadow="dark-lg"
      >
        <Heading mb={10}>Log in to LangChat</Heading>
        <Input
          value={username}
          onChange={(e) => setUsername(e.target.value)}
          onKeyDown={(e) => {
            if (e.key === "Enter") {
              signIn();
            }
          }}
          mb={5}
          placeholder="Username"
        />

        <InputGroup size="md">
          <Input
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            onKeyDown={(e) => {
              if (e.key === "Enter") {
                signIn();
              }
            }}
            mb={5}
            type={showPass ? "text" : "password"}
            placeholder="Password"
          />
          <InputRightElement width="4.5rem">
            <Button h="1.75rem" size="sm" onClick={handleHideToggle}>
              {showPass ? "Hide" : "Show"}
            </Button>
          </InputRightElement>
        </InputGroup>


        <Button type="submit" onClick={() => signIn()} colorScheme="blue">
          Login
        </Button>
      </Flex>
    </Flex>
  );
};

export default Login;
