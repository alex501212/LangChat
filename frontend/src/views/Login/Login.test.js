import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Login from "./Login";

afterEach(() => {
  cleanup();
});

it("matches snapshot", () => {
  const container = renderer.create(<Login />).toJSON();
  expect(container).toMatchSnapshot();
});

it("renders all text", () => {
  const view = render(<Login />);
  expect(screen.getByText("Log in to LangChat")).toBeInTheDocument();
  expect(screen.getByText("Show")).toBeInTheDocument();
  expect(screen.getByText("Login")).toBeInTheDocument();
  expect(screen.getByPlaceholderText("Username")).toBeInTheDocument();
  expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
});
