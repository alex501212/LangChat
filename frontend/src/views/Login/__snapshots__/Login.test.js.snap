// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-1muqdga"
>
  <div
    className="css-1pk8ame"
  >
    <h2
      className="chakra-heading css-1r2f04i"
    >
      Log in to LangChat
    </h2>
    <input
      className="chakra-input css-r7q7qr"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Username"
      value=""
    />
    <input
      className="chakra-input css-r7q7qr"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </div>
</div>
`;
